### DISCUSSION TITLE
A windows that automatically changes to show the driver cam with the closest gaps

### DISCUSSION THREAD

**Discussion by getoffmylawn18** [f1multiviewer/issue-tracker#68] [Feature requests]
I miss the old PLC three-box layout that typically showed the two closest on-track battles in the 2nd and 3rd windows. 

My idea would be to have an additional choice of driver cam windows that automatically switches between the closest on-track gaps. 

Maybe windows that automatically change between the 1st, 2nd, and 3rd closest gaps?

An automatic view might get a little chaotic during starts and safety cars, but right now I'm left manually looking at the data screen to identify the driver cam that is the most likely to make a pass. It would be nice if that feature could be automated.

**Comment by f1multiviewer** [comment]
This one is very similar to https://github.com/f1multiviewer/issue-tracker/discussions/63, so I think I'll close this one and hope you can vote on that one (or comment to add more details)

**Reply by getoffmylawn18** [reply]
Thank you for your reply!

Respectfully, what I'm asking for is very different from #63 even if the technical implementation is similar. 

If a gap to the car ahead is 0.5 seconds, a pass is imminent and that onboard should be "front and center." It doesn't matter if that is 2nd place fighting for 1st or 13th going for 12th.

On-track battles are the action everyone wants to see and those only happen when the gaps are small. A dynamic viewport would show all the important action regardless of where it is happening in the field if a window was programmed to show the onboards of the closest gaps. 

Most of the time, a fixed view of 2nd or 3rd place like #63 is requesting won't show anything interesting because the gap to the car ahead is so large the onboard doesn't even show the leading car. But once that gap falls under 1 second, that's when things are getting spicy and that onboard should be highlighted.  It doesn't matter if the battle is for 1st place or 12th. 

_Note: 
Many battles happen simultaneously (after a safety car, last lap, etc) and it could make sense to have multiple windows showing the 1st, 2nd, and 3rd closest gaps. If the views jump around too much between the 1-2-3 windows, code the logic so a viewport doesn't change if the gap stays within the top three closest gaps and that viewport only switches over if that gap falls outside of the top three and then is replaced by the new 3rd closest gap. 

Anyway, that is getting into the weeds a bit, but it would be one way to implement it._
